File: mod-main/src/main/kotlin/server/ApiServerConfig.kt
package com.idkidknow.mcreallink.server

import java.io.File

data class ApiServerConfig<TLanguage>(
    val port: Int,
    val language: TLanguage,
    val tlsConfig: TlsConfig,
)

sealed interface TlsConfig {
    object None : TlsConfig
    data class Tls(val certChain: File, val privateKey: File) : TlsConfig
    data class MutualTls(val certChain: File, val privateKey: File, val root: File) : TlsConfig
}

File: core/src/main/scala/com/idkidknow/mcreallink/server/ApiServerConfig.scala
package com.idkidknow.mcreallink.server

import com.idkidknow.mcreallink.lib.platform.Language
import fs2.io.file.Path

final case class ApiServerConfig[P[_]](
  port: Int,
  language: P[Language],
  TlsConfig: TlsConfig,
)

enum TlsConfig {
  case None
  case Tls(certChain: Path, privateKey: Path)
  case MutualTls(certChain: Path, privateKey: Path, root: Path)
}
